webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateJob.java

```diff
@@ -0,0 +1,257 @@
+                  return;
+                }
+                String sourceLang = repository.getSourceLocale().getBcp47Tag().split("-")[0];
+                if (isReuseSourceOnLanguageMatch
+                    && targetLocale.getBcp47Tag().startsWith(sourceLang)) {
+                  reuseSourceStringAsTranslation(tmTextUnit, repository, targetLocale, sourceLang);
+                  return;
+                }
+                // Get the prompt override for this locale if it exists, otherwise use the
+                // repository default
+                RepositoryLocaleAIPrompt repositoryLocaleAIPrompt =
+                    repositoryLocaleAIPrompts.get(targetLocale.getBcp47Tag()) != null
+                        ? repositoryLocaleAIPrompts.get(targetLocale.getBcp47Tag())
+                        : repositoryLocaleAIPrompts.get(REPOSITORY_DEFAULT_PROMPT);
+                if (repositoryLocaleAIPrompt != null && !repositoryLocaleAIPrompt.isDisabled()) {
+                  executeTranslationPrompt(
+                      tmTextUnit, repository, targetLocale, repositoryLocaleAIPrompt);
+                } else {
+                  logger.debug(
+                      "No active translation prompt found for locale: {}, skipping AI translation.",
+                      targetLocale.getBcp47Tag());
+                  meterRegistry.counter(
+                      "AITranslateJob.translate.noActivePrompt",
+                      Tags.of(
+                          "repository",
+                          repository.getName(),
+                          "locale",
+                          targetLocale.getBcp47Tag()));
+                }
+              } catch (Exception e) {
+                logger.error(
+                    "Error translating text unit id {} for locale: {}",
+                    tmTextUnit.getId(),
+                    targetLocale.getBcp47Tag(),
+                    e);
+                meterRegistry.counter(
+                    "AITranslateJob.translate.error",
+                    Tags.of(
+                        "repository", repository.getName(), "locale", targetLocale.getBcp47Tag()));
+              }
+            });
+  }
+
+  private boolean isExistingVariant(TMTextUnit tmTextUnit, Locale targetLocale) {
+    return tmTextUnitCurrentVariantRepository.findByLocale_IdAndTmTextUnit_Id(
+            targetLocale.getId(), tmTextUnit.getId())
+        != null;
+  }
+
+  private void reuseSourceStringAsTranslation(
+      TMTextUnit tmTextUnit, Repository repository, Locale targetLocale, String sourceLang) {
+    logger.debug(
+        "Target language {} matches source language {}, re-using source string as translation.",
+        targetLocale.getBcp47Tag(),
+        sourceLang);
+    meterRegistry.counter(
+        "AITranslateJob.translate.reuseSourceAsTranslation",
+        Tags.of("repository", repository.getName(), "locale", targetLocale.getBcp47Tag()));
+    tmService.addTMTextUnitVariant(
```

Comment:
readability nit: could we add some whitespace/an endline between the logging/metrics statements and the business logic (`tmService.addTMTextUnitVariant`)?